scenes/lessons/lesson/lesson.py

```diff
@@ -0,0 +1,165 @@
+from manim import Scene, MathTex, Write, VGroup, AnimationGroup, Line, GrowFromEdge, Circumscribe, Wiggle, Indicate, SurroundingRectangle, Create
+from manim import DOWN, LEFT, RIGHT, UP, DL
+from manim import RED, ORANGE, YELLOW, GREEN
+
+FONT_SIZE = 90
+
+class Lesson(Scene):
+    """Դասի նպատակն է բացատրել, թե բնական թվերի իրար տակ բազմապատկումը ոնց է բացատրվում բազմապատկման և գումարման բաշխական օրենքով։"""
+    def construct(self):
+        self.wait()
+
+        tex1 = MathTex("7 \\cdot 134 = ", "938", font_size=115)
+        tex1[-1].set_opacity(0)
+        tex1.to_edge(UP)
+        srr_rect = SurroundingRectangle(tex1, color=GREEN, buff=.2)
+        self.play(
+            Create(srr_rect),
+            Write(tex1)
+        )
+        self.wait()
+
+        tex2 = MathTex(
+            "7 \\cdot 134 = 7 \\cdot (100 + 30+ 4) =",
+            "7 \\cdot 100 + 7 \\cdot 30 + 7 \\cdot 4 =",
+            "700 + 210 + 28 =",
+            "700 + 200 + 10 + 20 + 8 =",
+            "(700 + 200) + (10 + 20) + 8 =",
+            "900 + 30 + 8 = 938",
+            font_size=60
+        )
+        
+        tex2.arrange(DOWN, aligned_edge=LEFT, buff=.4).to_edge(DL).shift(UP*.5)
+        
+        num1 = MathTex("134", font_size=FONT_SIZE)
+        multiply_sign = MathTex("\\times", font_size=FONT_SIZE-20)
+        num2 = MathTex("7", font_size=FONT_SIZE)
+
+        num2.next_to(num1, DOWN, aligned_edge=RIGHT)
+        multiply_sign.next_to(VGroup(num1, num2), LEFT)
+        VGroup(num1, num2, multiply_sign).to_edge(RIGHT, buff=1.6).shift(UP*1.9)
+        
+        line1 = Line().set_length(VGroup(num1, num2).width+.4)
+        line1.next_to(VGroup(num1, num2), DOWN)
+
+        first_number = tex2[2][8:10].copy()
+        second_number = tex2[2][4:7].copy()
+        third_number = tex2[2][0:3].copy()
+
+        for tex in tex2:
+            if tex == tex2[3]:
```

Comment:
enumerate֊ով վազել ու ինդեքս ստուգելը ավելի ապահով չէ՞
Թե՞ արժեքով ստուգելն էր քեզի կարևոր

Reply:
սկզբում if-երը չկային, մենակ Write(tex) էր դրա համար էի ըտենց գրել for-ը, հետո if-երի կարիքը զգացվեց ու էլ չփոխեցի, ինչ որ էական տարբերություն կա՞

Reply:
դե ==֊ը անհասկանալի է որոշ օբյեկտների համար, կարող է փորձեկը կետ առ կետ ստուգե նույնն է կոորդինատներով թե չէ։ Եթե ինդեքսները ստուգես, կարդալուց էլ է ավելի տրամաբանական, աշխատելուց էլ։

Մեկ էլ, անունները կլավացնե՞ս tex֊ը մի էլեմենտը, tex2-ը՝ լրիվը, քիչմ ակնհայտ չէ, տոր համար tex==tex2[3] տեսնելուց շատ կասկածելի էր